### PR TITLE
sign in & sign up view

### LINKED ISSUE #23: Login View
https://github.com/codeday-labs/bookworms/projects/1#card-65218978
https://github.com/codeday-labs/bookworms/projects/1#card-65219014

### PR DESCRIPTION
**Pull request by dknv**
having a sign in window and sign up page for (registered) users

### CODE REVIEW

**Comment by dknv** [comment]
I don't know why the test failed on here...it was able to find SignIn.js and passed on my laptop :/

**Comment by jeromewu** [comment]
> I don't know why the test failed on here...it was able to find SignIn.js and passed on my laptop :/

In my local environment it fails as well, let me take a look.

**Comment by dknv** [comment]
please do, and let me know!

**Comment by jeromewu** [comment]
This is a typical issue in Windows. Lol

If you check [this folder](https://github.com/auroravo2001/bookworms/tree/signin-view/client/src/components/form) in github, you will find the file name is `Signup` instead of `SignUp` (maybe in your local it is the correct name). That's why your program cannot find this file. The easy way to fix is just rename the file to other name and change it back to the one with correct upper letters.

**Comment by dknv** [comment]
ahhh ahahaha okay - got it. I'll do that for the files and see what happens :D

**Comment by dknv** [comment]
Yay! It worked :)

File: client/src/components/form/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          </Form.Group>
+
+          <br/>             
+
+          <Button type="submit" disabled={isPending}>  {isPending ? 'Signing in...' : 'Sign in'} </Button>
+        </Form>
+      </Modal.Body>
+
+      <Modal.Footer>
+        <Link to="/signup" onClick={() => handleClose()}> Not a registered user? </Link>
```
Comment (jeromewu) [inline client/src/components/form/SignIn.js]: You can write it like this:

```javascript
<Link to="/signup" onClick={handleClose}>
```

**Review by jeromewu — CHANGES_REQUESTED** [review]
Only one comment, well done. :smile:

File: client/src/components/form/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        }
+    },
+
+    twitter: {
+        backgroundColor: '#00acee',
+        '&:hover': {
+            backgroundColor: '#33bcf1',
+        }
+    }
+})
```
Comment (dknv) [inline client/src/components/form/SignUp.js:41]: I have a question -- why can't we put this in a css and use `styles.fb` or `styles.gmail` or so to style the different respective components? I tried but it doesn't work the same way, and I can't figure out why...isn't styling in here and writing these classes in a css file the same thing?

Comment (jeromewu) [reply client/src/components/form/SignUp.js:41]: It is a good question that requires few lines to explain:

For stylesheets, there are few common ways to do that:
- CSS
- Sass
- Less
- CSS in Javascript / [JSS](https://cssinjs.org/?v=v10.7.1)

For `import styles from './styles.css'`, you are actually using CSS and if you try to print something like `styles.author`, you will get a string **className** like `reviewList_author__26HZg` which is a auto generated class name base on your css file. (with some namespace and hash to make it unique)

For `makeStyles`, you are using CSS in Javascript / JSS, which aims to represent styles in JSON object, and convert to CSS   right before the moment you use it. So the output of `classes.fb` is a string like `makeStyles-fb-4` which is a object key to access JSON object representing the style.

So when you pass `styles.author` to material-ui components, the meaning of the string is different and they cannot recognize, so it cannot work.

You can check this link for why they are using JSS: https://material-ui.com/styles/basics/#why-use-material-uis-styling-solution

Comment (dknv) [reply client/src/components/form/SignUp.js:41]: ohhh, I see! Now that explains a lot -- that is also probably why I can't change the color of the material-ui components...Thank you!

**Comment by jeromewu** [comment]
Merged, thanks for the hard work. :smile:

**Pull request merged**